Tools/Scripts/libraries/webkitapipy/webkitapipy/macho.py

```
@@ -0,0 +1,166 @@
+                # Address in binary and symbol name:
+                # ```
+                # 0x00A5B75C  _JSBigIntCreateWithDouble
+                # ```
+                offset, symbol = line.split(maxsplit=1)
+                # skip the header line
+                if offset != 'offset':
+                    self.exports.add(symbol)
+            elif in_section == Sect.IMPORTS:
+                # Hexadecimal index, symbol name, optional linkage tag, and
+                # containing dylib as recorded in the two-level namespace:
+                # ```
+                # 0x02E0  _voucher_mach_msg_set [weak-import] (from libSystem)
+                # ```
+                idx, symbol, metadata = line.split(maxsplit=2)
+                dylib = metadata[metadata.index('(from ') + 6:-1]
+                if dylib != '<this-image>':
+                    self.imports.add(symbol)
+            elif in_section == Sect.OBJC:
+                # ObjC-like declaration for classes and protocols with method
+                # names and method addresses for classes:
+                # ```
+                # @interface WebEventRegion : NSObject <NSCopying>
+                #   0x02134AC4  -[WebEventRegion initWithPoints::::]
+                #   0x02134B80  -[WebEventRegion copyWithZone:]
+                #   0x02134BB8  -[WebEventRegion description]
+                #   0x02134C58  -[WebEventRegion hitTest:]
+                #   0x02134CD4  -[WebEventRegion isEqual:]
+                #   0x02134E94  -[WebEventRegion quad]
+                #   0x02134F74  -[WebEventRegion p1]
+                #   0x02134F9C  -[WebEventRegion p2]
+                #   0x02134FC4  -[WebEventRegion p3]
+                #   0x02134FEC  -[WebEventRegion p4]
+                # @end
+                # @protocol NSCopying :
+                #   -[NSCopying copyWithZone:]
+                # @end
+                # ```
+                m = method_to_selector.search(line)
+                if m:
```

Do we care about lines that don't match? Do we always want to silently ignore them? It feels like a change in output would potentially cause us to stop finding methods we should.

---

That's definitely a risk. Maybe we can complain about unrecognized output (similar to what I do at the bottom of the loop) when there is no match, and the line doesn't contain `@interface` / `@protocol` / `@end`.